File: vijaymamra/vijaymamra/custom_script/sales_invoice/sales_invoice.js
// Copyright (c) 2017, Frappe Technologies Pvt. Ltd. and Contributors
// License: GNU General Public License v3. See license.txt

frappe.ui.form.on("Sales Invoice Item", {
	quantity_of_bags: function(frm, cdt, cdn) {
		var child = locals[cdt][cdn];
		if(child){
			if (child.quantity_of_bags){
				frappe.call({
						method:"patanjali.patanjali.custom_script.sales_invoice.sales_invoice.calculate_row_quantity",
						args:{
							"bag_qty": child.quantity_of_bags,
							"rate": child.rate,
							"uom": child.conversion_factor
						},
						callback: function(r) {
							if(r.message) {
								console.log(r.message)
								frappe.model.set_value(cdt, cdn, "qty",r.message);
							}
						}
					})
			} 
		}
	}
	
});

File: vijaymamra/vijaymamra/custom_script/sales_invoice/sales_invoice.py
from __future__ import unicode_literals
import frappe
from frappe.model.document import Document
import decimal

@frappe.whitelist()
def calculate_row_quantity(bag_qty, rate, uom):
	bag = float(bag_qty)
	conv_factor = float(uom)
	row_qty = bag / conv_factor
	return round(row_qty)


def calculate_qty(doc, method=None):
	if doc.items:
		total_qty = 0
		bag_qty = 0
		for row in doc.items:
			total_qty = row.qty + total_qty
			bag_qty = row.quantity_of_bags + bag_qty
		doc.total_quantity_in_kg = total_qty
		doc.total_bags_quantity = bag_qty
